src/browser/plugins.ts

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2017 OpenFin Inc.
+
+Licensed under OpenFin Commercial License you may not use this file except in compliance with your Commercial License.
+Please contact OpenFin Inc. at sales@openfin.co to obtain a Commercial License.
+*/
+
+import * as path from 'path';
+import { getStartManifest, StartManifest } from './core_state';
+import { Plugin } from '../shapes';
+import { readFile } from 'fs';
+import { rvmMessageBus } from './rvm/rvm_message_bus';
+
+interface PluginWithContent extends Plugin {
+    _content: string;
+}
+
+/**
+ * A map of already-retrieved plugin module paths.
+ *
+ * Examples:
+ * 'pluginName: 1.0.0' -> 'path\to\plugin\injectable\script.js'
+ * 'pluginName2: 0.0.1' -> '' // plugin/version doesn't exist
+ */
+const pluginPaths: Map<string, string> = new Map();
+
+/**
+ * Gets all plugins defined in app's manifest
+ */
+export async function getModules(): Promise<PluginWithContent[]> {
+    const {url, data: {plugin: plugins = []}} = <StartManifest>getStartManifest();
+    const promises = plugins.map((plugin: Plugin) => getModule(url, plugin));
+    return await Promise.all(promises);
+}
+
+/**
+ * Gets a single plugin module
+ */
+async function getModule(sourceUrl: string, plugin: Plugin): Promise<PluginWithContent> {
```

Comment:
:neckbeard: you can avoid an early return here by `if/else` setting the plugin path

Reply:
Early `return` in this function prevents the following RVM request

Reply:
I know, but if you put it in the else it wont hit it either